examples/c/ex_storage_source.c

```diff
@@ -0,0 +1,1142 @@
+        goto err;
+    op = "size";
+    if ((ret = fh->fh_size(fh, session, &size)) != 0)
+        goto err;
+    op = "size-compare";
+    if ((size_t)size != len || (size_t)size > sizeof(buf)) {
+        ret = EINVAL;
+        goto err;
+    }
+    op = "read";
+    if ((ret = fh->fh_read(fh, session, 0, len, buf)) != 0)
+        goto err;
+    op = "read-compare";
+    if (strncmp(buf, content, len) != 0) {
+        ret = EINVAL;
+        goto err;
+    }
+
+err:
+    if (fh != NULL && (t_ret = fh->close(fh, session)) != 0 && ret == 0) {
+        op = "close";
+        ret = t_ret;
+    }
+    if (ret != 0)
+        fprintf(stderr, "demo failed during %s: %s\n", op, wiredtiger_strerror(ret));
+    else
+        printf("demo succeeded read %s\n", objname);
+
+    return (ret);
+}
+
+static int
+demo_test_list(WT_STORAGE_SOURCE *ss, WT_SESSION *session, WT_LOCATION_HANDLE *location,
+  const char *prefix, uint32_t limit, uint32_t expect)
+{
+    char **obj_list;
+    const char *op;
+    uint32_t i, obj_count;
+    int ret, t_ret;
+
+    obj_list = NULL;
+    op = "location_list";
+    if ((ret = ss->ss_location_list(ss, session, location, prefix, limit, &obj_list, &obj_count)) !=
+      0)
+        goto err;
+    op = "location_list count";
+    if (obj_count != expect) {
+        ret = EINVAL;
+        goto err;
+    }
+    printf("list: %s:\n", (const char *)location);
+    for (i = 0; i < obj_count; i++) {
+        printf("  %s\n", obj_list[i]);
+    }
+
+err:
+    if (obj_list != NULL &&
+      (t_ret = ss->ss_location_list_free(ss, session, obj_list, obj_count)) != 0 && ret == 0) {
+        op = "location_list_free";
+        ret = t_ret;
+    }
+    if (ret != 0)
+        fprintf(stderr, "demo failed during %s: %s\n", op, wiredtiger_strerror(ret));
+    else
+        printf("demo succeeded location_list %s\n", (const char *)location);
+
+    return (ret);
+}
+
+static int
+demo_test_storage_source(WT_STORAGE_SOURCE *ss, WT_SESSION *session)
+{
+    WT_LOCATION_HANDLE *location1, *location2;
+    const char *op;
+    int ret, t_ret;
+    bool exist;
+
+    location1 = location2 = NULL;
+    op = "location_handle";
+    if ((ret = ss->ss_location_handle(ss, session, "location-one", &location1)) != 0)
+        goto err;
+    if ((ret = ss->ss_location_handle(ss, session, "location-two", &location2)) != 0)
+        goto err;
+
+    /* Create and existence checks. */
+    op = "create/exist checks";
+    if ((ret = demo_test_create(ss, session, location1, "A", "location-one-A")) != 0)
+        goto err;
+
+    if ((ret = ss->ss_exist(ss, session, location1, "A", &exist)) != 0)
+        goto err;
+    if (!exist) {
+        fprintf(stderr, "Exist test failed for A\n");
+        ret = EINVAL;
+        goto err;
+    }
+    if ((ret = ss->ss_exist(ss, session, location2, "A", &exist)) != 0)
+        goto err;
+    if (exist) {
+        fprintf(stderr, "Exist test failed for A in location2\n");
+        ret = EINVAL;
+        goto err;
+    }
+
```

Comment:
Please add some comments about what we're testing here. In particular, why you chose `location-two-AA` instead of say, `location-two-C` and anything else of interest.